src/main/java/UseCases/ResponseImpl.java

```diff
@@ -0,0 +1,34 @@
+package UseCases;
+
+import Entities.Ingredient;
+import Entities.Recipe;
+
+import java.util.List;
+
+public class ResponseImpl<T> {
+    List<T> data;
```

Comment:
You've not inherited the proper parent class and so don't see the errors you have regarding these generic types and your function definitions.

Reply:
going to fix this up, and the other inheritance issue.